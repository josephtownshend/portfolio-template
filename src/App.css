.App {
  text-align: center;
}

.NavBar {
  min-height: 10vh;
}

/* .SiteTitle {
  padding: 20px;
  float: left;
  font-size: 20px;
} */

.SubMenu {
  padding: 20px;
  float: right;
  font-size: 20px;
}

.Button {
  border: none;
  padding: 20px;
  font-size: 20px;

}

.Button:focus {
  outline:0;
}

.Button_title {
padding: 20px;
float: left;
font-size: 20px;
border: none;

}

.Button_title:focus {
  outline:0;
}

.Button_subMenu {
  padding: 20px;
  float: right;
  font-size: 20px;
  border: none;
}

.Button_subMenu:focus {
  outline:0;
}

.CV h2 {
  font-size: 20px;
  font-weight: normal;
}

.Writing h2 {
  font-size: 20px;
  font-weight: normal;
}

.List {
  list-style-type: none;
  font-weight: normal;
  padding: 0px;
  line-height: 1.7em;
}

p.BlockText {
  margin-left: auto;
  margin-right: auto;
  width: 40em;
}
