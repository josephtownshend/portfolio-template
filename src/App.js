import React from 'react';
import Slideshow from './components/Slideshow';
import Header from './components/Header';
import Writing from './components/Writing';
import About from './components/About';
import './App.css';

export default class App extends React.Component {

  state = {
    whichComponent: "Slideshow"
  };

  handleChangeTitle = () => {
    this.setState({
      whichComponent: "Slideshow"
    });
  };

  handleChangeWriting = () => {
    this.setState({
      whichComponent: "Writing"
    });
  };

  handleChangeAbout = () => {
    this.setState({
      whichComponent: "About"
    });
  };

  render() {
    if (this.state.whichComponent === "Slideshow") {
      return (
        <div className="App">
          <Header
            whichComponent={this.state.whichComponent}
            handleChangeAbout={this.handleChangeAbout}
            handleChangeWriting={this.handleChangeWriting}
            handleChangeTitle={this.handleChangeTitle}
          />
          <Slideshow />
        </div>
      );
    } else if (this.state.whichComponent === "Writing") {
      return (
        <div className="App">
          <Header />
          <Writing />
        </div>
      );
    } else if (this.state.whichComponent === "About") {
      return (
        <div className="App">
          <Header />
          <About />
        </div>
      );
    }
    return null;
  }
}
