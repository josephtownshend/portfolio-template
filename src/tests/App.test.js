import React from 'react';
import ReactDOM from 'react-dom';
import { configure } from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from '../App';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow renders without crashing', () => {
  shallow(<App />)
});

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
