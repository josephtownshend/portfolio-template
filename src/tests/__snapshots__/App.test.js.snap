// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="NavBar"
  >
    <button
      className="Button_title"
      onClick={[Function]}
    >
      Artist Website
    </button>
    <button
      className="Button_subMenu"
      onClick={[Function]}
    >
      Contact
    </button>
    <button
      className="Button_subMenu"
      onClick={[Function]}
    >
      About
    </button>
    <button
      className="Button_subMenu"
      onClick={[Function]}
    >
      Writing
    </button>
  </div>
  <div>
    <img
      alt="Lauren Keeley"
      src="http://www.fruttagallery.com/wp-content/uploads/1.LaurenKeeley_InaYear_2015_Frutta.jpg"
      style={
        Object {
          "width": 900,
        }
      }
    />
  </div>
  <div>
    <button
      className="Button"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="Button"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
